### PR TITLE
Fix hardcoded CU count in MLA/indexer Triton kernels

### PR DESCRIPTION
**Pull request by samremes**
The DeepSeek MQA-logits indexer kernels defaulted TotalCuCount to 80 (an MI308-sized value) on gfx942.

Centralize get_num_sms() on chip_info.get_cu_num() (honors the CU_NUM env override and parses rocminfo) with a multi_processor_count fallback, and replace the three hardcoded TotalCuCount defaults in the indexer with None resolved at call time via get_num_sms(). Explicit caller overrides still take precedence.

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

File: aiter/ops/triton/utils/device_info.py [hunk 1]
```diff
@@ -1,12 +1,22 @@
-import torch
+import functools
 
 
+@functools.lru_cache(maxsize=1)
 def get_num_sms():
```
[the remark this answers is not in the document]
Comment (samremes) [reply aiter/ops/triton/utils/device_info.py:5]: Well, the device won't change.

**Review by valarLip — APPROVED** [review]
